openmc/data/decay.py

```diff
@@ -0,0 +1,458 @@
+    modes : list
+        Decay mode information for each mode of decay.
+    nuclide : dict
+        Dictionary describing decaying nuclide with keys 'name',
+        'excited_state', 'mass', 'stable', 'spin', and 'parity'.
+    spectra : dict
+        Resulting radiation spectra for each radiation type.
+
+    """
+    def __init__(self, ev_or_filename):
+        # Get evaluation if str is passed
+        if isinstance(ev_or_filename, Evaluation):
+            ev = ev_or_filename
+        else:
+            ev = Evaluation(ev_or_filename)
+        #assert ev.info['sublibrary'] == 'Radioactive decay data'
+
+        file_obj = StringIO(ev.section[8, 457])
+
+        self.nuclide = {}
+        self.modes = []
+        self.spectra = {}
+        self.average_energies = {}
+
+        # Get head record
+        items = get_head_record(file_obj)
+        Z, A = divmod(items[0], 1000)
+        metastable = items[3]
+        self.nuclide['atomic_number'] = Z
+        self.nuclide['mass_number'] = A
+        self.nuclide['isomeric_state'] = metastable
+        if metastable > 0:
+            self.nuclide['name'] = '{}{}_m{}'.format(ATOMIC_SYMBOL[Z], A,
+                                                    metastable)
+        else:
+            self.nuclide['name'] = '{}{}'.format(ATOMIC_SYMBOL[Z], A)
+        self.nuclide['mass'] = items[1]  # AWR
+        self.nuclide['excited_state'] = items[2]  # State of the original nuclide
+        self.nuclide['stable'] = (items[4] == 1)  # Nucleus stability flag
+
+        # Determine if radioactive/stable
+        if not self.nuclide['stable']:
+            NSP = items[5]  # Number of radiation types
+
+            # Half-life and decay energies
+            items, values = get_list_record(file_obj)
+            self.half_life = ufloat(items[0], items[1])
+            NC = items[4]//2
+            pairs = [x for x in zip(values[::2], values[1::2])]
+            ex = self.average_energies
+            ex['light'] = ufloat(*pairs[0])
+            ex['electromagnetic'] = ufloat(*pairs[1])
+            ex['heavy'] = ufloat(*pairs[2])
+            if NC == 17:
+                ex['beta-'] = ufloat(*pairs[3])
+                ex['beta+'] = ufloat(*pairs[4])
+                ex['auger'] = ufloat(*pairs[5])
+                ex['conversion'] = ufloat(*pairs[6])
+                ex['gamma'] = ufloat(*pairs[7])
+                ex['xray'] = ufloat(*pairs[8])
+                ex['Bremsstrahlung'] = ufloat(*pairs[9])
+                ex['annihilation'] = ufloat(*pairs[10])
+                ex['alpha'] = ufloat(*pairs[11])
+                ex['recoil'] = ufloat(*pairs[12])
+                ex['SF'] = ufloat(*pairs[13])
+                ex['neutron'] = ufloat(*pairs[14])
+                ex['proton'] = ufloat(*pairs[15])
+                ex['neutrino'] = ufloat(*pairs[16])
+
+            items, values = get_list_record(file_obj)
+            spin = items[0]
+            if spin == -77.777:
```

Comment:
-77.777 can't be represented exactly in single precision.  Will this equality check work?